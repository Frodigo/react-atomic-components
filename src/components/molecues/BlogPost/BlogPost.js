import React from 'react';
import Moment from 'react-moment';
import PropTypes from 'prop-types';

import Link from '../../atoms/Link';

import classes from './BlogPost.module.css';

const BlogPost = (props) => {
    const {
        title,
        author,
        date,
        url,
        image,
        shortDescription,
        ...rest
    } = props;

    return (
        <div className={classes.block} {...rest}>
            <div className={classes.blockInner}>
                <section className={classes.text}>
                    <h2 className={classes.heading}>
                        <a href={url}>{title}</a>
                    </h2>

                    <div className={classes.meta}>
                        <span className={classes.author}>{author}</span>
                        <span className={classes.metaItem}>
                            <Moment
                                format="D MMMM YYYY"
                                date={new Date(date)}
                            />
                        </span>
                    </div>

                    <Link variant="primary" href={url}>
                        Read more
                    </Link>
                </section>
                <div className={classes.shortDescription}>
                    <p>
                        {shortDescription}{' '}
                        <a href={url}>Read more</a>
                    </p>
                </div>
                <figure className={classes.figure}>
                    <a href={url}>
                        <img src={image} alt={title} width={500} />
                    </a>
                </figure>
            </div>
        </div>
    );
};

BlogPost.propTypes = {
    author: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    shortDescription: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};

export default BlogPost;
