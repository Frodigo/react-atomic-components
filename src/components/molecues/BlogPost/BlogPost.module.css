.block {
    display: flex;
    flex-direction: column;
    padding: 30px 0;
    border-bottom: solid 2px var(--color-text);
}

.blockInner {
    padding: 0 var(--spacing-xs);
}

.heading {
    margin-top: 0;
    margin-bottom: 12px;
}
.heading a {
    color: inherit;
    font-size: inherit;
    text-decoration: none;
}

.meta {
    margin-bottom: 12px;
}

.metaItem {
    display: inline-block;
    margin-left: 17px;
}

.author {
    composes: metaItem;
    font-weight: 700;
}

.metaItem:first-child {
    margin-left: 0;
}

.figure {
    display: none;
}

.shortDescription {
    display: none;
}


@media (min-width: 768px) {
    .meta + a {
        display: none;
    }

    .blockInner {
        align-items: center;
        display: flex;
    }

    .text,
    .figure  {
        flex: 1 1 50%;
    }

    .figure {
        display: flex;
        justify-content: flex-end;
        padding-left: var(--spacing-md);
    }
}

@media (min-width: 1300px) {
    .blockInner {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
    }

    .heading a:hover,
    .heading a:focus {
        color: var(--color-secondary)
    }

    .shortDescription {
        display: block;
        padding: 0 var(--spacing-md);
    }

    .text {
        padding-right: var(--spacing-md);
    }

    .text,
    .figure,
    .shortDescription  {
        flex: 1 1 33%;
    }
}