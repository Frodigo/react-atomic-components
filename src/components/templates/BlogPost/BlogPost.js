import React from 'react';

import PageHeader from '../../organisms/PageHeader/PageHeader';
import Breadcrumbs from '../../molecues/Breadcrumbs';
import Newsletter from '../../organisms/Newsletter';
import PageFooter from '../../organisms/PageFooter/PageFooter';
import MobileNav from '../../organisms/MobileNav/MobileNav';
import { items as mobileNavItems } from '../../organisms/MobileNav/MobileNav.config';
import Heading from '../../atoms/Text/Heading';
import BlockQuote from '../../atoms/Text/Blockquote/Blockquote';
import AboutTheAuthor from '../../organisms/AboutTheAuthor';
import ShareSection from '../../organisms/ShareSection';
import Tout from '../../molecues/Tout';

import classes from './BlogPost.module.css';

const BlogPost = () => {
    const breadcrumbsItems = [
        {
            label: 'Home',
            url: '#'
        },
        {
            label: 'Blog',
            url: '#'
        },
        {
            label: 'PWA Studio',
            url: '#'
        }
    ];

    const newsletterProps = {
        title: 'Subscribe to the newsletter',
        subtitle: 'I will send you unique content every Friday.',
        submitUrl: 'https://app.getresponse.com/add_subscriber.html',
        token: 'WMdq9',
        thankYouPageUrl: 'https://marcin-kwiatkowski.com/thank-you',
        footer:
            'Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers',
        onSubmit: (data, e) => e.target.submit()
    };

    const post = {
        author: 'John',
        date: 'November 27, 2020',
        heroImage: 'https://via.placeholder.com/320x180',
        title: 'Heading level 1'
    };

    const aboutTheAuthorProps = {
        name: 'John',
        image: 'https://via.placeholder.com/270x270',
        description:
            'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.',
        ctaUrl: '#',
        ctaLabel: 'Read more'
    };

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home',
            isActive: true
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    return (
        <>
            <PageHeader navItems={pageHeaderNavItems} />
            <div className={classes.postHeader}>
                <Heading type="h1">{post.title}</Heading>
                <div className={classes.postMetadata}>
                    {post.author} / {post.date}
                </div>
                <Breadcrumbs items={breadcrumbsItems} />
            </div>

            <figure className={classes.postHeroImage}>
                <img src={post.heroImage} alt={post.title} />
            </figure>

            <div className={classes.content}>
                <p>
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio. Vestibulum interdum vestibulum felis
                    ac molestie.
                </p>
                <h2>Heading Level 2</h2>

                <p>
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio. Vestibulum interdum vestibulum felis
                    ac molestie.
                </p>
                <h3>Heading Level 3</h3>

                <figure>
                    <img
                        src="https://via.placeholder.com/320x180"
                        alt="figure"
                    />
                </figure>

                <h4>Heading level 4</h4>
                <p className="paragraphSmaller">
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio. Vestibulum interdum vestibulum felis
                    ac molestie.
                </p>

                <BlockQuote
                    author="Marcin Kwiatkowski"
                    quote="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin tempus ullamcorper ante, 
                    eu mattis est auctor id. Nunc libero nisl, efficitur at aliquet vel, venenatis eget felis. Suspendisse at diam nec 
                    risus scelerisque tempus. Mauris id eros diam. Mauris ac risus id libero laoreet eleifend in ac elit."
                />

                <p className="paragraphLarge">
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio. Vestibulum interdum vestibulum felis
                    ac molestie.
                </p>

                <h5>Heading level 5</h5>

                <ol>
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ol>

                <h6>Heading level 6</h6>

                <ul>
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ul>
            </div>

            <AboutTheAuthor {...aboutTheAuthorProps} />
            <Newsletter {...newsletterProps} />
            <ShareSection />
            <div className={classes.relatedProducts}>
                <Heading type="h4">You may also like</Heading>
                <div className={classes.items}>
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                    />
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                    />
                </div>
            </div>
            <PageFooter footerColumns={[
                {
                    header: 'Column 1',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 2',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 3',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 4',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                }
            ]} footerBottomLinks={[
                {
                    href: '#',
                    label: 'Link 1'
                },
                {
                    href: '#',
                    label: 'Link 2'
                },
                {
                    href: '#',
                    label: 'Link 3'
                }
            ]} copyright="&copy; 2021 Lorem ipsum" />
            <MobileNav items={mobileNavItems} />
        </>
    );
};

export default BlogPost;
