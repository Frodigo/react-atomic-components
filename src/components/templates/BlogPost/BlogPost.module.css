.postHeader {
    padding: var(--spacing-md);
}

.postMetadata {
    border-bottom: 2px solid var(--color-secondary);
    margin-bottom: var(--spacing-md);
    padding: var(--spacing-xs) 0;
}

.postHeader nav {
    border-bottom: 0;
    margin-top: 5px;
    padding: 0
}

.postHeroImage {
    margin-bottom: var(--spacing-xxl);
}

.postHeroImage img {
    display: block;
    width: 100%;
}



.relatedProducts h4 {
    margin: 30px 0;
    text-align: center;
}

@media (min-width: 768px) {
    .items {
        display: flex;
        padding-left: 15px;
        padding-right: 15px;
    }

    .relatedProducts {
        padding-bottom: var(--spacing-xxl);
    }

    .items > div {
        flex: 1 1 50%;
        margin: 0 var(--spacing-md);
    }
}

@media (min-width: 1300px) {
    .postHeader,
    .postHeroImage {
        margin: 0 auto;
        max-width: var(--small-desktop-container-width);
        padding-left: 0;
        padding-right: 0;
    }

    .postHeader {
        margin-top: 40px;
    }

    .content {
       margin-top: 40px;

    }

    .content > div:first-child {
        max-width: var(--small-desktop-container-width);
    }

    .relatedProducts {
        max-width: var(--desktop-container-width);
        margin: 60px auto;
    }
}
