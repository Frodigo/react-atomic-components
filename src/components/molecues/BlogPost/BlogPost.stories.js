import React from 'react';

import { default as BlogPostComponent } from './BlogPost';

export default {
    title: 'Molecues',
    component: BlogPostComponent
};

const Template = (args) => <BlogPostComponent {...args} />;

export const BlogPost = Template.bind({});
BlogPost.args = {
    title: 'Lorem ipsum dolor sit amet, vel maximus ante faucibus sed.',
    author: 'Jan Kowalski',
    date: 'October 10, 2020',
    url: '#',
    image: 'https://via.placeholder.com/396x222',
    shortDescription: 'Quisque rutrum ac lectus ullamcorper placerat. Morbi condimentum malesuada dui, id consectetur lectus dictum vel. Integer bibendum mauris id nisl dignissim, ut lacinia dolor commodo. Sed eros nisl, interdum hendrerit felis ut, pellentesque mattis eros. '
};
