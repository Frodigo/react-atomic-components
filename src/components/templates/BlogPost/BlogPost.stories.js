import React from 'react';

import { default as BlogPostComponent } from './BlogPost';

export default {
    title: 'Templates',
    component: BlogPostComponent
};

const Template = (args) => <BlogPostComponent {...args} />;

export const BlogPost = Template.bind({});
