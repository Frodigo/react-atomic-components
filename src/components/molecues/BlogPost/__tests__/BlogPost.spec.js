import React from 'react';
import { render } from '@testing-library/react';

import BlogPost from '../BlogPost';

describe('Blog post should be rendered', () => {
    const titleMock = 'This is a heading';
    const authorMock = 'Lorem ipsum';
    const dateMock = 'October 10, 2020';
    const urlMock = '#';
    const imageMock = 'https://via.placeholder.com/396x222';
    const shortDescriptionMock = 'Lorem ipsum';

    test('Blog post renders correctly', () => {
        const { container } = render(
            <BlogPost
                title={titleMock}
                author={authorMock}
                date={dateMock}
                url={urlMock}
                image={imageMock}
                shortDescription={shortDescriptionMock}
            />
        );

        expect(container).toMatchSnapshot();
    });
});
