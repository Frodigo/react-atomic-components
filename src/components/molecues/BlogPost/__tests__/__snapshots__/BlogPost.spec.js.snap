// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog post should be rendered Blog post renders correctly 1`] = `
<div>
  <div
    class="block"
  >
    <div
      class="blockInner"
    >
      <section
        class="text"
      >
        <h2
          class="heading"
        >
          <a
            href="#"
          >
            This is a heading
          </a>
        </h2>
        <div
          class="meta"
        >
          <span
            class="author"
          >
            Lorem ipsum
          </span>
          <span
            class="metaItem"
          >
            <time
              datetime="1602280800000"
            >
              10 October 2020
            </time>
          </span>
        </div>
        <a
          class="primary"
          href="#"
        >
          Read more
        </a>
      </section>
      <div
        class="shortDescription"
      >
        <p>
          Lorem ipsum
           
          <a
            href="#"
          >
            Read more
          </a>
        </p>
      </div>
      <figure
        class="figure"
      >
        <a
          href="#"
        >
          <img
            alt="This is a heading"
            src="https://via.placeholder.com/396x222"
            width="500"
          />
        </a>
      </figure>
    </div>
  </div>
</div>
`;
